src/cloudant/result.py

```diff
@@ -71,7 +71,14 @@ def python_to_couch(options):
         if key not in ARG_TYPES:
             msg = 'Invalid argument {0}'.format(key)
             raise CloudantArgumentError(msg)
-        if not isinstance(val, ARG_TYPES[key]):
+        # pylint: disable=unidiomatic-typecheck
+        if (
+                not isinstance(val, ARG_TYPES[key]) or
+                (
```

Comment:
I think this `or` could use a comment for explanation, to avoid anyone breaking this additional check without really understanding why it exists (although the test I think would fail anyway I think a comment would be worth having).